### PR TITLE
POTEL 40 - Init Priority settings

### PR DESCRIPTION
**Pull request by adinauer**
## :scroll: Description

Sometimes Sentry.init is called multiple times, these settings allow us to control which call to init takes priority.

## :bulb: Motivation and Context

Support different re-init scenarios as well as init calls coming from multiple integrations (e.g. OTel Agent, Logging Integration, Spring Boot)

## :pencil: Checklist

- [ ] I reviewed the submitted code.
- [ ] I added tests to verify the changes.
- [ ] No new PII added or SDK only sends newly added PII if `sendDefaultPII` is enabled.
- [ ] I updated the docs if needed.
- [ ] Review from the native team if needed.
- [ ] No breaking change or entry added to the changelog.
- [ ] No breaking change for hybrid SDKs or communicated to hybrid SDKs.

### CODE REVIEW

File: sentry/src/test/java/io/sentry/SentryOptionsTest.kt
Comment (lbloder) [inline sentry/src/test/java/io/sentry/SentryOptionsTest.kt:1]: Maybe add a test here for the default value of `InitPriority`:

```
    @Test
    fun `when options is initialized, InitPriority is set to MEDIUM by default`() {
        assertEquals(SentryOptions().initPriority, InitPriority.MEDIUM)
    }
```

File: sentry/src/test/java/io/sentry/util/InitUtilTest.kt [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+import io.sentry.InitPriority
+import io.sentry.SentryOptions
+import kotlin.test.Test
+import kotlin.test.assertFalse
+import kotlin.test.assertTrue
+
+class InitUtilTest {
+
+    var previousOptions: SentryOptions? = null
```
Comment (lbloder) [inline sentry/src/test/java/io/sentry/util/InitUtilTest.kt:11]: These could be private.

Also, should we re-init them for each test with `@Before`?

File: sentry/src/main/java/io/sentry/SentryOptions.java [hunk 2]
```diff
@@ -2562,6 +2592,7 @@ public SentryOptions() {
    */
   private SentryOptions(final boolean empty) {
     if (!empty) {
+      setInitPriority(InitPriority.LOWEST);
```
[the remark this answers is not in the document]
Comment (adinauer) [reply sentry/src/main/java/io/sentry/SentryOptions.java:2595]: Was intended differently, let's remove this.

**Review by lbloder — APPROVED** [review]
LGTM 👍

**Review by markushi — APPROVED** [review]

**Pull request merged**